### ISSUE TITLE
checkout/finish is not sending order confirmation email

### ISSUE THREAD

**Issue by danielDigitalArt** [Adyen/adyen-shopware5#120]
**Describe the bug**
When the Payment is processed and the Checkout-Finish Page shows the Order-Data no Orderconfirmation Mail is sent.

**To Reproduce**
1. Go to "Checkout"
2. select a payment method (i.e. Ideal from Adyen)
3. Do the normal process
4. see the checkout screen with the order data, no email is received

**Expected behavior**
The Order-Confirmation Mail is sent through the normal Email Process, as other Plugins might also have registered to the corresponding Event.

**Desktop (please complete the following information):**
 - OS: OSX
 - Browser Chrome
 - Version 88

**Additional context**
Shopversion: 5.6.9
Pluginversion: 1.6.0 (from Shopware Store)
Plugin is running in Test-Mode.
Emails are generally working, when using the normal Prepayment Method the email gets sent.

Other Plugins which might alter the behavior of the Shop in the Checkout-Process were deactivated to verifiy if the problem might be coming from them, but the problem persisted. Other Plugins were a Shopware-Language-Plugin installed and a Plugin for having a deliverydate selectable in the Checkout process.

**Comment by acampos1916** [comment]
Hello @danielDigitalArt,

The plugin indeed will send the email only after a successful payment per this pull request https://github.com/Adyen/adyen-shopware5/pull/46. I assume that your tests were for a successful payment? If so could you try step-debugging with Xdebug the code from that PR to find the problem?

Ángel

**Comment by acampos1916** [comment]
Possible duplicate of #110 @danielDigitalArt could you verify if you're experiencing the same with the orderId=null?

**Comment by danielDigitalArt** [comment]
@acampos1916 how would i force the shop to use OrderID null instead of letting the shop insert its own orderID?

**Comment by acampos1916** [comment]
@danielDigitalArt no sorry, I didn't mean that you should force the value to null, but rather checking if the orderId is null just like in that issue to see if you're experiencing the same.

**Comment by danielDigitalArt** [comment]
@acampos1916 i tried to find out if there is an orderID set or not, but i was only able to install the profiler plugin, but i could not see any variable here for "orderID" or "order.id" so i do not really know how i can test this properly. Problem is, that this system i already installed on a server where i can not add xdebug or read the code with an IDE, while debugging this. i Could try to add some vardumps into the adyen files, but that could also change the behavior and makes it hard to test properly.

**Comment by acampos1916** [comment]
Hello @danielDigitalArt,

Could you check if #125 solves the issue for you? We found out that the OrderId is not present at that moment of the order flow as you suggested, so a fallback to the ordernumber was added.

**Comment by danielDigitalArt** [comment]
@acampos1916 I tried to clone the plugin into the AdyenPayment folder, the i ran the composer command and it went fine so far, in the backend i could then click the update button and can see the Version 1.7.0. When im trying to make an order i can not see a payment method. In the Shop-Log i can see this error:

> Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException: You have requested a non-existent service "shop". Did you mean one of these: "adyen_payment.components.payload.providers.shopper_info_provider", "adyen_payment.components.shopware_version_check", "customer_search.dbal.condition.ordered_in_shop_handler", "customer_search.dbal.condition.registered_in_shop_handler", "shop_page_menu", "shopware.api.address", "shopware.api.article", "shopware.api.cache", "shopware.api.category", "shopware.api.country", "shopware.api.customer", "shopware.api.customer_group", "shopware.api.customer_stream", "shopware.api.emotion_preset", "shopware.api.manufacturer", "shopware.api.media", "shopware.api.order", "shopware.api.payment_methods", "shopware.api.property_group", "shopware.api.shop", "shopware.api.translation", "shopware.api.user", "shopware.api.variant", "shopware.auth.constraint.user_email_validator", "shopware.auth.constraint.user_name_validator", "shopware.auth.validator.user_validator", "shopware.backend.global_search", "shopware.benchmark_bundle.benchmark_statistics", "shopware.benchmark_bundle.bi_transmission", "shopware.benchmark_bundle.client", "shopware.benchmark_bundle.collector", "shopware.benchmark_bundle.commands.receive_data", "shopware.benchmark_bundle.commands.send_data", "shopware.benchmark_bundle.components.template_caching_handler", "shopware.benchmark_bundle.components.translation", "shopware.benchmark_bundle.hydrator.statistics_response_hydrator", "shopware.benchmark_bundle.matcher_payment", "shopware.benchmark_bundle.matcher_shipping", "shopware.benchmark_bundle.providers.analytics", "shopware.benchmark_bundle.providers.categories", "shopware.benchmark_bundle.providers.customers", "shopware.benchmark_bundle.providers.emotions", "shopware.benchmark_bundle.providers.manufacturers", "shopware.benchmark_bundle.providers.orders", "shopware.benchmark_bundle.providers.payments", "shopware.benchmark_bundle.providers.plugins", "shopware.benchmark_bundle.providers.products", "shopware.benchmark_bundle.providers.shipments", "shopware.benchmark_bundle.providers.shop", "shopware.benchmark_bundle.providers.shopware", "shopware.benchmark_bundle.providers.updated_orders", "shopware.benchmark_bundle.repository.config", "shopware.benchmark_bundle.statistics_transmission", "shopware.cache_manager", "shopware.captcha.default_captcha", "shopware.captcha.honeypot_captcha", "shopware.captcha.legacy_captcha", "shopware.captcha.no_captcha", "shopware.captcha.repository", "shopware.captcha.validator", "shopware.cart.basket_helper", "shopware.cart.basket_query_helper", "shopware.cart.net_rounding", "shopware.cart.net_rounding.after_quantity", "shopware.cart.net_rounding.after_tax", "shopware.cart.proportional_cart_merger", "shopware.cart.proportional_tax_calculator", "shopware.commands.admin_create_command", "shopware.commands.cache_clear_command", "shopware.commands.clone_category_tree_command", "shopware.commands.cron_list_command", "shopware.commands.cron_run_command", "shopware.commands.database_setup_command", "shopware.commands.first_run_wizard_disable_command", "shopware.commands.first_run_wizard_enable_command", "shopware.commands.generate_attributes_command", "shopware.commands.generate_product_feed_command", "shopware.commands.list_product_feed_command", "shopware.commands.make_migration_command", "shopware.commands.migrate_article_attribute_translations_command", "shopware.commands.migrations_migrate_command", "shopware.commands.plugin_activate_command", "shopware.commands.plugin_config_list_command", "shopware.commands.plugin_config_set_command", "shopware.commands.plugin_deactivate_command", "shopware.commands.plugin_delete_command", "shopware.commands.plugin_install_command", "shopware.commands.plugin_list_command", "shopware.commands.plugin_refresh_command", "shopware.commands.plugin_reinstall_command", "shopware.commands.plugin_uninstall_command", "shopware.commands.plugin_update_command", "shopware.commands.rebuild_category_tree_command", "shopware.commands.rebuild_seo_index_command", "shopware.commands.refresh_search_index_command", "shopware.commands.refresh_topseller_command", "shopware.commands.session_cleanup_command", "shopware.commands.settings_labels_find_missing_command", "shopware.commands.snippets_find_missing_command", "shopware.commands.snippets_remove_command", "shopware.commands.snippets_to_db_command", "shopware.commands.snippets_to_ini_command", "shopware.commands.snippets_to_sql_command", "shopware.commands.snippets_validate_command", "shopware.commands.store_download_command", "shopware.commands.store_list_command", "shopware.commands.store_list_domains_command", "shopware.commands.store_list_integrated_command", "shopware.commands.store_list_updates_command", "shopware.commands.stream_index_populate_command", "shopware.commands.theme_cache_generate_command", "shopware.commands.theme_create_command", "shopware.commands.theme_dump_configuration_command", "shopware.commands.theme_initialize_command", "shopware.commands.theme_synchronize_command", "shopware.commands.thumbnail_cleanup_command", "shopware.commands.thumbnail_generate_command", "shopware.commands.warm_up_http_cache_command", "shopware.components.cart.cart_migration", "shopware.components.cart.cart_migration_cleaner", "shopware.components.cart.proportional_cart_normalizer", "shopware.components.dispatch_format_helper", "shopware.components.last_articles_subscriber", "shopware.components.optin_cleanup_service", "shopware.components.optin_service", "shopware.components.privacy.cookie_remove_subscriber", "shopware.components.privacy.ip_anonymizer", "shopware.components.privacy.privacy_cleanup_cron_subscriber", "shopware.components.privacy.privacy_service", "shopware.components.registration_cleanup_service", "shopware.components.shop_registration_service", "shopware.components.state_translator", "shopware.components.stream_protocol_validator", "shopware.components.template.html_minifier", "shopware.components.validator.order_number_constraint_validator", "shopware.console.completion_command", "shopware.csrftoken_validator", "shopware.customer_stream.cookie_subscriber", "shopware.customer_stream.criteria_factory", "shopware.customer_stream.repository", "shopware.customer_stream.stream_indexer", "shopware.emotion.emotion_exporter", "shopware.emotion.emotion_importer", "shopware.emotion.emotion_landingpage_loader", "shopware.emotion.emotion_presetdata_transformer", "shopware.emotion.preset_banner_component_handler", "shopware.emotion.preset_bannerslider_component_handler", "shopware.emotion.preset_categoryteaser_component_handler", "shopware.emotion.preset_data_synchronizer", "shopware.emotion.preset_html5_video_component_handler", "shopware.emotion.preset_installer", "shopware.emotion.preset_loader", "shopware.emotion.translation_importer", "shopware.emotion_component_installer", "shopware.escaper", "shopware.esd.public.url_generator", "shopware.filesystem.private", "shopware.filesystem.public", "shopware.filesystem.public.url_generator", "shopware.filesystem_factory", "shopware.form.constraint.exists", "shopware.form.constraint.unique", "shopware.form.extension.enlight", "shopware.form.extension.event", "shopware.form.factory", "shopware.form.string_renderer_service", "shopware.generic_controller_listener", "shopware.holiday_table_updater", "shopware.http_cache.cache_route_generation_service", "shopware.http_cache.cache_time_service", "shopware.http_cache.default_route_service", "shopware.http_cache.invalidation_date.blog_date", "shopware.http_cache.invalidation_date.blog_listing", "shopware.http_cache.invalidation_date.campaign_date", "shopware.http_cache.invalidation_date.listing_date", "shopware.http_cache.invalidation_date.listing_date_frontend", "shopware.http_cache.invalidation_date.product_date", "shopware.http_cache.route_installer", "shopware.log.fileparser", "shopware.logaware_reflection_helper", "shopware.model.search_builder", "shopware.number_range_incrementer", "shopware.openssl_verificator", "shopware.plugin.cached_config_reader", "shopware.plugin.config_reader", "shopware.plugin.config_writer", "shopware.plugin_payment_installer", "shopware.plugin_requirement_validator", "shopware.plugin_xml_plugin_reader", "shopware.query_operator_validator", "shopware.release", "shopware.requirements", "shopware.routing.generators.default_generator", "shopware.routing.generators.rewrite_generator", "shopware.routing.matchers.default_matcher", "shopware.routing.matchers.event_matcher", "shopware.routing.matchers.rewrite_matcher", "shopware.routing.post_filter.default_post_filter", "shopware.routing.post_filter.frontend_post_filter", "shopware.routing.pre_filter.default_pre_filter", "shopware.routing.pre_filter.frontend_pre_filter", "shopware.slug", "shopware.snippet_database_handler", "shopware.snippet_query_handler", "shopware.snippet_validator", "shopware.upload_max_size_validator", "shopware_account.address_service", "shopware_account.address_validator", "shopware_account.constraint.current_password_validator", "shopware_account.constraint.customer_email_validator", "shopware_account.constraint.form_email_validator", "shopware_account.constraint.password_validator", "shopware_account.customer_service", "shopware_account.customer_unlock_service", "shopware_account.customer_validator", "shopware_account.form.addressform", "shopware_account.form.attributeform", "shopware_account.form.emailupdateform", "shopware_account.form.passwordupdateform", "shopware_account.form.personalform", "shopware_account.form.profile_update_form", "shopware_account.form.resetpasswordform", "shopware_account.register_service", "shopware_account.store_front_greeting_service", "shopware_account.type.salutation_type", "shopware_attribute.backend_controllers_attribute_data", "shopware_attribute.backend_controllers_attributes", "shopware_attribute.backend_controllers_benchmark", "shopware_attribute.backend_controllers_benchmark_local_overview", "shopware_attribute.backend_controllers_benchmark_overview", "shopware_attribute.backend_controllers_entity_search", "shopware_attribute.blog_reader", "shopware_attribute.blog_repository", "shopware_attribute.blog_searcher", "shopware_attribute.category_reader", "shopware_attribute.category_repository", "shopware_attribute.category_searcher", "shopware_attribute.crud_service", "shopware_attribute.custom_facet_reader", "shopware_attribute.custom_facet_repository", "shopware_attribute.custom_facet_searcher", "shopware_attribute.customer_reader", "shopware_attribute.customer_repository", "shopware_attribute.customer_stream_reader", "shopware_attribute.customer_stream_repository", "shopware_attribute.customer_stream_searcher", "shopware_attribute.data_loader", "shopware_attribute.data_persister", "shopware_attribute.emotion_reader", "shopware_attribute.emotion_repository", "shopware_attribute.emotion_searcher", "shopware_attribute.media_reader", "shopware_attribute.media_repository", "shopware_attribute.media_searcher", "shopware_attribute.order_reader", "shopware_attribute.order_repository", "shopware_attribute.premium_reader", "shopware_attribute.premium_repository", "shopware_attribute.premium_searcher", "shopware_attribute.product_reader", "shopware_attribute.product_repository", "shopware_attribute.property_option_reader", "shopware_attribute.property_option_repository", "shopware_attribute.property_option_searcher", "shopware_attribute.repository_registry", "shopware_attribute.schema_operator", "shopware_attribute.shop_reader", "shopware_attribute.shop_repository", "shopware_attribute.shop_searcher", "shopware_attribute.table_mapping", "shopware_attribute.type_mapping", "shopware_backend_searchdbal.search_term_helper", "shopware_bundle.es_backend_subscriber.orm_backlog_subscriber", "shopware_bundle.search_bundle_dbal_condition_handler.variant_condition_handler", "shopware_bundle.search_bundle_dbal_facet_handler.variant_facet_handler", "shopware_bundle_emotion.component_handler.content_type_component_handler", "shopware_bundle_es_backend.backlog_service", "shopware_bundle_es_backend.commands.clear_backlog_command", "shopware_bundle_es_backend.subscriber.domain_backlog_subscriber", "shopware_bundle_esindexing.product.product_manual_position_loader", "shopware_bundle_search.criteria_request_handler.variant_criteria_request_handler", "shopware_bundle_sitemap.commands.sitemap_generate_command", "shopware_bundle_sitemap.config_handler.file", "shopware_bundle_sitemap.cronjob_subscriber", "shopware_bundle_sitemap.provider.url.blog", "shopware_bundle_sitemap.provider.url.category", "shopware_bundle_sitemap.provider.url.custom", "shopware_bundle_sitemap.provider.url.landingpage", "shopware_bundle_sitemap.provider.url.manufacturer", "shopware_bundle_sitemap.provider.url.product", "shopware_bundle_sitemap.provider.url.static", "shopware_bundle_sitemap.service.config_handler", "shopware_bundle_sitemap.service.filesystem", "shopware_bundle_sitemap.service.filter_container_factory", "shopware_bundle_sitemap.service.sitemap_exporter", "shopware_bundle_sitemap.service.sitemap_lister", "shopware_bundle_sitemap.service.sitemap_lock", "shopware_bundle_sitemap.service.sitemap_name_generator", "shopware_bundle_sitemap.service.sitemap_writer", "shopware_bundle_sitemap.service.url_filter", "shopware_bundle_sitemap.sitemap_controller", "shopware_bundle_sitemap.url_filter.blog", "shopware_bundle_sitemap.url_filter.category", "shopware_bundle_sitemap.url_filter.landing_page", "shopware_bundle_sitemap.url_filter.manufacturer", "shopware_bundle_sitemap.url_filter.product", "shopware_bundle_sitemap.url_filter.statics", "shopware_bundle_store_front.gateway_dbal.list_product_query_helper", "shopware_cache_warmer.url_provider_blog", "shopware_cache_warmer.url_provider_category", "shopware_cache_warmer.url_provider_emotion", "shopware_cache_warmer.url_provider_factory", "shopware_cache_warmer.url_provider_manufacturer", "shopware_cache_warmer.url_provider_product", "shopware_cache_warmer.url_provider_productwithcategory", "shopware_cache_warmer.url_provider_productwithnumber", "shopware_cache_warmer.url_provider_static", "shopware_cache_warmer.url_provider_variantswitch", "shopware_controllers_api_addresses", "shopware_controllers_api_articles", "shopware_controllers_api_caches", "shopware_controllers_api_categories", "shopware_controllers_api_countries", "shopware_controllers_api_customer_groups", "shopware_controllers_api_customer_streams", "shopware_controllers_api_customers", "shopware_controllers_api_error", "shopware_controllers_api_generate_article_images", "shopware_controllers_api_manufacturers", "shopware_controllers_api_media", "shopware_controllers_api_orders", "shopware_controllers_api_payment_methods", "shopware_controllers_api_property_groups", "shopware_controllers_api_shops", "shopware_controllers_api_translations", "shopware_controllers_api_users", "shopware_controllers_api_variants", "shopware_controllers_api_version", "shopware_controllers_backend_address", "shopware_controllers_backend_analytics", "shopware_controllers_backend_application", "shopware_controllers_backend_article", "shopware_controllers_backend_articlelist", "shopware_controllers_backend_articlepricevariation", "shopware_controllers_backend_banner", "shopware_controllers_backend_base", "shopware_controllers_backend_blog", "shopware_controllers_backend_cache", "shopware_controllers_backend_canceledorder", "shopware_controllers_backend_category", "shopware_controllers_backend_config", "shopware_controllers_backend_corelicense", "shopware_controllers_backend_csrftoken", "shopware_controllers_backend_customer", "shopware_controllers_backend_customerquickview", "shopware_controllers_backend_customerstream", "shopware_controllers_backend_customfacet", "shopware_controllers_backend_customsorting", "shopware_controllers_backend_document", "shopware_controllers_backend_emotion", "shopware_controllers_backend_emotionpreset", "shopware_controllers_backend_error", "shopware_controllers_backend_export", "shopware_controllers_backend_extjs", "shopware_controllers_backend_feedback", "shopware_controllers_backend_firstrunwizard", "shopware_controllers_backend_firstrunwizardpluginmanager", "shopware_controllers_backend_form", "shopware_controllers_backend_index", "shopware_controllers_backend_log", "shopware_controllers_backend_logger", "shopware_controllers_backend_login", "shopware_controllers_backend_mail", "shopware_controllers_backend_manual_sorting", "shopware_controllers_backend_mediamanager", "shopware_controllers_backend_newsletter", "shopware_controllers_backend_newslettermanager", "shopware_controllers_backend_notification", "shopware_controllers_backend_order", "shopware_controllers_backend_overview", "shopware_controllers_backend_partner", "shopware_controllers_backend_payment", "shopware_controllers_backend_performance", "shopware_controllers_backend_premium", "shopware_controllers_backend_productfeed", "shopware_controllers_backend_productstream", "shopware_controllers_backend_property", "shopware_controllers_backend_riskmanagement", "shopware_controllers_backend_search", "shopware_controllers_backend_shipping", "shopware_controllers_backend_site", "shopware_controllers_backend_snippet", "shopware_controllers_backend_supplier", "shopware_controllers_backend_systeminfo", "shopware_controllers_backend_theme", "shopware_controllers_backend_translation", "shopware_controllers_backend_updatewizard", "shopware_controllers_backend_userconfig", "shopware_controllers_backend_usermanager", "shopware_controllers_backend_vote", "shopware_controllers_backend_voucher", "shopware_controllers_backend_widgets", "shopware_controllers_frontend_account", "shopware_controllers_frontend_address", "shopware_controllers_frontend_ajaxsearch", "shopware_controllers_frontend_blog", "shopware_controllers_frontend_campaign", "shopware_controllers_frontend_checkout", "shopware_controllers_frontend_compare", "shopware_controllers_frontend_csrftoken", "shopware_controllers_frontend_custom", "shopware_controllers_frontend_detail", "shopware_controllers_frontend_error", "shopware_controllers_frontend_forms", "shopware_controllers_frontend_index", "shopware_controllers_frontend_listing", "shopware_controllers_frontend_media", "shopware_controllers_frontend_newsletter", "shopware_controllers_frontend_note", "shopware_controllers_frontend_register", "shopware_controllers_frontend_robotstxt", "shopware_controllers_frontend_search", "shopware_controllers_frontend_sitemap", "shopware_controllers_frontend_sitemapxml", "shopware_controllers_frontend_tellafriend", "shopware_controllers_frontend_tracking", "shopware_controllers_widgets_blog", "shopware_controllers_widgets_campaign", "shopware_controllers_widgets_captcha", "shopware_controllers_widgets_checkout", "shopware_controllers_widgets_compare", "shopware_controllers_widgets_emotion", "shopware_controllers_widgets_error", "shopware_controllers_widgets_index", "shopware_controllers_widgets_listing", "shopware_controllers_widgets_recommendation", "shopware_core.license_service_subscriber", "shopware_core.local_license_unpack_service", "shopware_customer_search.gateway.address_gateway", "shopware_customer_search.gateway.address_hydrator", "shopware_customer_search.gateway.customer_gateway", "shopware_customer_search.gateway.customer_hydrator", "shopware_elastic_search.backlog_processor", "shopware_elastic_search.backlog_reader", "shopware_elastic_search.client", "shopware_elastic_search.commands.analyze_command", "shopware_elastic_search.commands.backlog_clear_command", "shopware_elastic_search.commands.backlog_sync_command", "shopware_elastic_search.commands.index_cleanup_command", "shopware_elastic_search.commands.index_populate_command", "shopware_elastic_search.commands.switch_alias_command", "shopware_elastic_search.console.console_evaluation_helper", "shopware_elastic_search.domain_backlog_subscriber", "shopware_elastic_search.field_mapping", "shopware_elastic_search.identifier_selector", "shopware_elastic_search.index_factory", "shopware_elastic_search.orm_backlog_save_subscriber", "shopware_elastic_search.orm_backlog_subscriber", "shopware_elastic_search.product_configuration_loader", "shopware_elastic_search.product_indexer", "shopware_elastic_search.product_listing_variation_loader", "shopware_elastic_search.product_mapping", "shopware_elastic_search.product_provider", "shopware_elastic_search.product_query_factory", "shopware_elastic_search.product_synchronizer", "shopware_elastic_search.property_indexer", "shopware_elastic_search.property_mapping", "shopware_elastic_search.property_provider", "shopware_elastic_search.property_query_factory", "shopware_elastic_search.property_synchronizer", "shopware_elastic_search.shop_analyzer", "shopware_elastic_search.shop_indexer", "shopware_elastic_search.shop_indexer_factory", "shopware_elastic_search.text_mapping", "shopware_elastic_search.text_mapping_factory", "shopware_emotion.component_handler.article", "shopware_emotion.component_handler.article_slider", "shopware_emotion.component_handler.banner", "shopware_emotion.component_handler.banner_slider", "shopware_emotion.component_handler.blog", "shopware_emotion.component_handler.category_teaser", "shopware_emotion.component_handler.event_fallback", "shopware_emotion.component_handler.html5video", "shopware_emotion.component_handler.html_code", "shopware_emotion.component_handler.html_element", "shopware_emotion.component_handler.iframe", "shopware_emotion.component_handler.manufacturer_slider", "shopware_emotion.component_handler.youtube", "shopware_emotion.data_collection_resolver", "shopware_emotion.emotion_element_gateway", "shopware_emotion.emotion_element_hydrator", "shopware_emotion.emotion_element_service", "shopware_emotion.emotion_gateway", "shopware_emotion.emotion_hydrator", "shopware_emotion.emotion_service", "shopware_emotion.emotion_struct_converter", "shopware_emotion.store_front_emotion_device_configuration", "shopware_es_backend.commands.index_cleanup_command", "shopware_es_backend.commands.index_populate_command", "shopware_es_backend.commands.sync_backlog_command", "shopware_es_backend.customer_searcher", "shopware_es_backend.indexer", "shopware_es_backend.order_searcher", "shopware_es_backend.product_searcher", "shopware_es_backend.search_query_builder", "shopware_media.adapter.ftp", "shopware_media.adapter.gcp", "shopware_media.adapter.local", "shopware_media.adapter.s3", "shopware_media.cdn_optimizer_service", "shopware_media.commands.image_migrate", "shopware_media.commands.media_cleanup", "shopware_media.commands.media_optimize", "shopware_media.extension_mapping", "shopware_media.garbage_collector", "shopware_media.garbage_collector_factory", "shopware_media.media_migration", "shopware_media.media_service", "shopware_media.media_service_factory", "shopware_media.optimizer.guetzli", "shopware_media.optimizer.jpegoptim", "shopware_media.optimizer.jpegtran", "shopware_media.optimizer.optipng", "shopware_media.optimizer.pngcrush", "shopware_media.optimizer.pngout", "shopware_media.optimizer_service", "shopware_media.replace_service", "shopware_media.service_subscriber", "shopware_media.strategy", "shopware_media.strategy_factory", "shopware_plugininstaller.account_manager_service", "shopware_plugininstaller.http_client", "shopware_plugininstaller.legacy_plugin_installer", "shopware_plugininstaller.plugin_download_service", "shopware_plugininstaller.plugin_extractor", "shopware_plugininstaller.plugin_installer", "shopware_plugininstaller.plugin_installer_struct_hydrator", "shopware_plugininstaller.plugin_licence_service", "shopware_plugininstaller.plugin_manager", "shopware_plugininstaller.plugin_service_local", "shopware_plugininstaller.plugin_service_store_production", "shopware_plugininstaller.plugin_service_view", "shopware_plugininstaller.store_client", "shopware_plugininstaller.store_order_service", "shopware_plugininstaller.subscription_service", "shopware_plugininstaller.unique_id_generator", "shopware_product_stream.criteria_factory", "shopware_product_stream.facet_filter", "shopware_product_stream.repository", "shopware_search.batch_product_number_search", "shopware_search.batch_product_search", "shopware_search.category_tree_facet_result_builder", "shopware_search.core_criteria_request_handler", "shopware_search.custom_facet_criteria_request_handler", "shopware_search.product_number_search", "shopware_search.property_criteria_request_handler", "shopware_search.search_term_pre_processor", "shopware_search.sorting_criteria_request_handler", "shopware_search.store_front_criteria_factory", "shopware_search.variant_search", "shopware_search_es.category_condition_handler", "shopware_search_es.category_facet_handler", "shopware_search_es.closeout_condition_handler", "shopware_search_es.combined_condition_facet_handler", "shopware_search_es.combined_condition_handler", "shopware_search_es.combined_condition_query_builder", "shopware_search_es.create_date_condition_handler", "shopware_search_es.customer_group_condition_handler", "shopware_search_es.handler_collection", "shopware_search_es.has_pseudo_price_condtion_handler", "shopware_search_es.height_condition_handler", "shopware_search_es.immediate_delivery_condition_handler", "shopware_search_es.immediate_delivery_facet_handler", "shopware_search_es.is_available_condition_handler", "shopware_search_es.is_new_condtion_handler", "shopware_search_es.last_id_condition_handler", "shopware_search_es.length_condition_handler", "shopware_search_es.main_variant_condition_handler", "shopware_search_es.manual_sorting_handler", "shopware_search_es.manufacturer_condition_handler", "shopware_search_es.manufacturer_facet_handler", "shopware_search_es.ordernumber_condition_handler", "shopware_search_es.popularity_sorting_handler", "shopware_search_es.price_condition_handler", "shopware_search_es.price_facet_handler", "shopware_search_es.price_field_mapper", "shopware_search_es.price_sorting_handler", "shopware_search_es.product_attribute_condition_handler", "shopware_search_es.product_attribute_facet_handler", "shopware_search_es.product_attribute_sorting_handler", "shopware_search_es.product_dimensions_facet_handler", "shopware_search_es.product_name_sorting_handler", "shopware_search_es.product_number_search_factory", "shopware_search_es.product_number_sorting_handler", "shopware_search_es.product_stock_sorting_handler", "shopware_search_es.productid_condition_handler", "shopware_search_es.property_condition_handler", "shopware_search_es.property_facet_handler", "shopware_search_es.release_date_condition_handler", "shopware_search_es.release_date_sorting_handler", "shopware_search_es.sales_condition_handler", "shopware_search_es.search_random_sorting_handler", "shopware_search_es.search_ranking_sorting_handler", "shopware_search_es.search_term_condition_handler", "shopware_search_es.search_term_query_builder", "shopware_search_es.service_subscriber", "shopware_search_es.shipping_free_condition_handler", "shopware_search_es.shipping_free_facet_handler", "shopware_search_es.similar_product_condition_handler", "shopware_search_es.struct_hydrator", "shopware_search_es.variant_condition_facet_handler", "shopware_search_es.variant_condition_handler", "shopware_search_es.variant_search", "shopware_search_es.vote_average_condition_handler", "shopware_search_es.vote_average_facet_handler", "shopware_search_es.weight_condition_handler", "shopware_search_es.width_condition_handler", "shopware_searchdbal.cache_keyword_finder", "shopware_searchdbal.category_condition_handler_dbal", "shopware_searchdbal.category_facet_handler_dbal", "shopware_searchdbal.closeout_condition_handler_dbal", "shopware_searchdbal.combined_condition_facet_handler_dbal", "shopware_searchdbal.combined_condition_handler_dbal", "shopware_searchdbal.create_date_condition_handler", "shopware_searchdbal.customer_group_condition_handler_dbal", "shopware_searchdbal.dbal_query_builder_factory", "shopware_searchdbal.has_pseudo_price_condition_handler_dbal", "shopware_searchdbal.height_condition_handler", "shopware_searchdbal.immediate_delivery_condition_handler_dbal", "shopware_searchdbal.immediate_delivery_facet_handler_dbal", "shopware_searchdbal.is_available_condition_handler_dbal", "shopware_searchdbal.is_new_condition_handler_dbal", "shopware_searchdbal.keyword_finder_dbal", "shopware_searchdbal.last_id_condition_handler_dbal", "shopware_searchdbal.length_condition_handler", "shopware_searchdbal.listing_price_helper", "shopware_searchdbal.listing_price_switcher", "shopware_searchdbal.listing_price_table", "shopware_searchdbal.manual_sorting_handler", "shopware_searchdbal.manufacturer_condition_handler_dbal", "shopware_searchdbal.manufacturer_facet_handler_dbal", "shopware_searchdbal.ordernumber_condition_handler_dbal", "shopware_searchdbal.popularity_sorting_handler_dbal", "shopware_searchdbal.price_condition_handler_dbal", "shopware_searchdbal.price_facet_handler_dbal", "shopware_searchdbal.price_sorting_handler_sorting_handler_dbal", "shopware_searchdbal.product_attribute_condition_handler_dbal", "shopware_searchdbal.product_attribute_facet_handler_dbal", "shopware_searchdbal.product_dimensions_facet_handler", "shopware_searchdbal.product_name_sorting_handler_dbal", "shopware_searchdbal.product_number_sorting_handler_dbal", "shopware_searchdbal.product_stock_sorting_handler_dbal", "shopware_searchdbal.productid_condition_handler_dbal", "shopware_searchdbal.property_condition_handler_dbal", "shopware_searchdbal.property_facet_handler_dbal", "shopware_searchdbal.random_sorting_handler_dbal", "shopware_searchdbal.release_date_condition_handler", "shopware_searchdbal.release_date_sorting_handler_dbal", "shopware_searchdbal.sales_condition_handler_dbal", "shopware_searchdbal.search_condition_handler_dbal", "shopware_searchdbal.search_indexer", "shopware_searchdbal.search_price_helper_dbal", "shopware_searchdbal.search_query_builder_dbal", "shopware_searchdbal.search_ranking_sorting_handler_dbal", "shopware_searchdbal.search_term_helper", "shopware_searchdbal.search_term_logger", "shopware_searchdbal.shipping_free_condition_handler_dbal", "shopware_searchdbal.shipping_free_facet_handler_dbal", "shopware_searchdbal.shopware_searchdbal.product_attribute_sorting_handler_dbal", "shopware_searchdbal.similar_products_handler_dbal", "shopware_searchdbal.variant_helper", "shopware_searchdbal.vote_average_condition_handler_dbal", "shopware_searchdbal.vote_average_facet_handler_dbal", "shopware_searchdbal.weight_condition_handler", "shopware_searchdbal.width_condition_handler", "shopware_static_content.gateway.dbal.esd_gateway", "shopware_static_content.service.download_service", "shopware_static_content.service.esd_service", "shopware_storefront.additional_text_service", "shopware_storefront.attribute_hydrator_dbal", "shopware_storefront.base_product_factory", "shopware_storefront.blog_gateway_dbal", "shopware_storefront.blog_hydrator_dbal", "shopware_storefront.blog_service", "shopware_storefront.cached_href_lang_service", "shopware_storefront.category_depth_service", "shopware_storefront.category_gateway", "shopware_storefront.category_hydrator_dbal", "shopware_storefront.category_service", "shopware_storefront.cheapest_price_es_gateway", "shopware_storefront.cheapest_price_es_service", "shopware_storefront.cheapest_price_gateway", "shopware_storefront.cheapest_price_service", "shopware_storefront.configurator_gateway", "shopware_storefront.configurator_hydrator_dbal", "shopware_storefront.configurator_options_gateway", "shopware_storefront.configurator_service", "shopware_storefront.context_service", "shopware_storefront.country_gateway", "shopware_storefront.country_hydrator_dbal", "shopware_storefront.currency_gateway_dbal", "shopware_storefront.currency_hydrator_dbal", "shopware_storefront.custom_facet_gateway", "shopware_storefront.custom_facet_service", "shopware_storefront.custom_listing_hydrator", "shopware_storefront.custom_sorting_gateway", "shopware_storefront.custom_sorting_service", "shopware_storefront.customer_group_gateway", "shopware_storefront.customer_group_hydrator_dbal", "shopware_storefront.download_gateway", "shopware_storefront.download_hydrator_dbal", "shopware_storefront.esd_hydrator_dbal", "shopware_storefront.field_helper_dbal", "shopware_storefront.graduated_prices_gateway", "shopware_storefront.graduated_prices_service", "shopware_storefront.href_lang_service", "shopware_storefront.link_gateway", "shopware_storefront.link_hydrator_dbal", "shopware_storefront.list_product_gateway", "shopware_storefront.list_product_service", "shopware_storefront.listing_link_rewrite_service", "shopware_storefront.locale_hydrator_dbal", "shopware_storefront.location_service", "shopware_storefront.manufacturer_gateway", "shopware_storefront.manufacturer_hydrator_dbal", "shopware_storefront.manufacturer_service", "shopware_storefront.marketing_service", "shopware_storefront.media_gateway", "shopware_storefront.media_hydrator_dbal", "shopware_storefront.media_service", "shopware_storefront.payment_gateway", "shopware_storefront.payment_hydrator_dbal", "shopware_storefront.price_calculation_service", "shopware_storefront.price_calculator", "shopware_storefront.price_group_discount_gateway", "shopware_storefront.price_hydrator_dbal", "shopware_storefront.product_configuration_gateway", "shopware_storefront.product_download_service", "shopware_storefront.product_hydrator_dbal", "shopware_storefront.product_link_service", "shopware_storefront.product_media_gateway", "shopware_storefront.product_number_service", "shopware_storefront.product_property_gateway", "shopware_storefront.product_service", "shopware_storefront.product_stream_hydrator_dbal", "shopware_storefront.property_gateway", "shopware_storefront.property_hydrator_dbal", "shopware_storefront.property_service", "shopware_storefront.related_product_streams_gateway", "shopware_storefront.related_product_streams_service", "shopware_storefront.related_products_gateway", "shopware_storefront.related_products_service", "shopware_storefront.shop_gateway_dbal", "shopware_storefront.shop_hydrator_dbal", "shopware_storefront.shop_page_children_gateway", "shopware_storefront.shop_page_gateway", "shopware_storefront.shop_page_hydrator_dbal", "shopware_storefront.shop_page_service", "shopware_storefront.similar_products_gateway", "shopware_storefront.similar_products_service", "shopware_storefront.storefront_cache", "shopware_storefront.tax_gateway", "shopware_storefront.tax_hydrator_dbal", "shopware_storefront.template_hydrator_dbal", "shopware_storefront.unit_hydrator_dbal", "shopware_storefront.variant_cheapest_price_gateway", "shopware_storefront.variant_cover_service", "shopware_storefront.variant_listing_price_service", "shopware_storefront.variant_media_gateway", "shopware_storefront.vote_average_gateway", "shopware_storefront.vote_gateway", "shopware_storefront.vote_hydrator_dbal", "shopware_storefront.vote_service", "shopware.api.customergroup", "shopware.api.customerstream", "shopware.api.emotionpreset", "shopware.api.propertygroup", "shopware.db", "shopware.event_manager", "shopware.hook_manager", "shopware.loader", "shopware.locale", "shopware.mail_transport", "shopware.model_annotations", "shopware.model_config", "shopware.model_manager", "shopware.plugin_manager", "shopware.plugin_xml_plugin_info_reader", "shopware_attribute.customer_searcher", "shopware_attribute.order_searcher", "shopware_attribute.product_searcher", "shopware_elastic_search.client.logger", "shopware_plugininstaller.plugin_service_store", "shopware_search.product_search", "shopware_search_es.product_number_search", "shop"? in /vendor/symfony/dependency-injection/Container.php:348 Stack trace:
> #0 /engine/Shopware/Components/DependencyInjection/Container.php(220): Symfony\Component\DependencyInjection\Container->get('shop', 1)
> #1 /engine/Shopware/Components/DependencyInjection/Container.php(141): Shopware\Components\DependencyInjection\Container->doLoad('shop', 'shop', 1)
> #2 /engine/Shopware/Components/DependencyInjection/Bridge/Session.php(82): Shopware\Components\DependencyInjection\Container->get('shop')
> #3 /var/www/vhosts/remondis-nl.digital-art.de/httpdocs3/var/cache/production_202011060821/proxies/ShopwareProductionfbf75f235570f0ff891379cf983f5dc277d15eb3ProjectContainer.php(2863): Shopware\Components\DependencyInjection\Bridge\Session->createSession(Object(ShopwareProductionfbf75f235570f0ff891379cf983f5dc277d15eb3ProjectContainer), Object(Shopware\Components\Session\PdoSessionHandler))
> #4 /vendor/symfony/dependency-injection/Container.php(306): ShopwareProductionfbf75f235570f0ff891379cf983f5dc277d15eb3ProjectContainer->getSessionService()
> #5 /engine/Shopware/Components/DependencyInjection/Container.php(220): Symfony\Component\DependencyInjection\Container->get('session', 1)
> #6 /engine/Shopware/Components/DependencyInjection/Container.php(141): Shopware\Components\DependencyInjection\Container->doLoad('session', 'session', 1)
> #7 /var/www/vhosts/remondis-nl.digital-art.de/httpdocs3/var/cache/production_202011060821/proxies/ShopwareProductionfbf75f235570f0ff891379cf983f5dc277d15eb3ProjectContainer.php(6126): Shopware\Components\DependencyInjection\Container->get('session')
> #8 /vendor/symfony/dependency-injection/Container.php(306): ShopwareProductionfbf75f235570f0ff891379cf983f5dc277d15eb3ProjectContainer->getAdyenPayment_Components_Adyen_Payment_MethodService()
> #9 /engine/Shopware/Components/DependencyInjection/Container.php(220): Symfony\Component\DependencyInjection\Container->get('adyen_payment.c...', 1)
> #10 /engine/Shopware/Components/DependencyInjection/Container.php(141): Shopware\Components\DependencyInjection\Container->doLoad('adyen_payment.c...', 'adyen_payment.c...', 1)
> #11 /engine/Library/Enlight/Controller/Action.php(413): Shopware\Components\DependencyInjection\Container->get('adyen_payment.c...')
> #12 /custom/plugins/AdyenPayment/Controllers/Backend/TestAdyenApi.php(10): Enlight_Controller_Action->get('adyen_payment.c...')
> #13 /engine/Library/Enlight/Controller/Action.php(192): Shopware_Controllers_Backend_TestAdyenApi->runAction()
> #14 /engine/Library/Enlight/Controller/Dispatcher/Default.php(478): Enlight_Controller_Action->dispatch('runAction')
> #15 /engine/Library/Enlight/Controller/Front.php(228): Enlight_Controller_Dispatcher_Default->dispatch(Object(Enlight_Controller_Request_RequestHttp), Object(Enlight_Controller_Response_ResponseHttp))
> #16 /engine/Shopware/Kernel.php(188): Enlight_Controller_Front->dispatch()
> #17 /vendor/symfony/http-kernel/HttpCache/SubRequestHandler.php(102): Shopware\Kernel->handle(Object(Symfony\Component\HttpFoundation\Request), 1, true)
> #18 /vendor/symfony/http-kernel/HttpCache/HttpCache.php(453): Symfony\Component\HttpKernel\HttpCache\SubRequestHandler::handle(Object(Shopware\Kernel), Object(Symfony\Component\HttpFoundation\Request), 1, true)
> #19 /engine/Shopware/Components/HttpCache/AppCache.php(261): Symfony\Component\HttpKernel\HttpCache\HttpCache->forward(Object(Symfony\Component\HttpFoundation\Request), true, NULL)
> #20 /vendor/symfony/http-kernel/HttpCache/HttpCache.php(238): Shopware\Components\HttpCache\AppCache->forward(Object(Symfony\Component\HttpFoundation\Request), true)
> #21 /engine/Shopware/Components/HttpCache/AppCache.php(102): Symfony\Component\HttpKernel\HttpCache\HttpCache->pass(Object(Symfony\Component\HttpFoundation\Request), true)
> #22 /var/www/vhosts/remondis-nl.digital-art.de/httpdocs3/shopware.php(122): Shopware\Components\HttpCache\AppCache->handle(Object(Symfony\Component\HttpFoundation\Request))
> #23

Im not really sure what is the problem here, i always had trouble when i tried to use the Adyen Payment Plugin from Github, im running this on a normal Shopware installation on a webserver that i installed for testing puproses, as it is it could be set to live immediately by removing its Basic Auth protection, so i dont think that the Problem comes from the Installation itself. Also i do not have any Problems when using the Version from the Shopware-Store.

**Comment by acampos1916** [comment]
Hello @danielDigitalArt,

Thanks for sharing this log. I see there that the TestAdyenApi.php is in use, but this is only accessible through the plugin's config panel when clicking on the Test button. I don't think those logs are related to the issue you're having.

Could you check if there's an error in the browser console when the methods are not shown? Also, if in dev mode you should be able to check the Adyen logs in `var/log` to see more info.

**Comment by danielDigitalArt** [comment]
@acampos1916 Yes, then it was when i tried if the credentials still work, which was not responding. I checked the logs, but there is absolutely nothing else from today. I now switched back to the version 1.6 from the shopware-Store and clicked reinstall. After this the Payment Methods were showing normally.

![Bildschirmfoto 2021-02-02 um 10 38 50](https://user-images.githubusercontent.com/25100967/106581185-e1794480-6542-11eb-8cc5-cc1778dd51c6.png)

When it didnt work it showed "Betaalwijze wijzigen" without any Payment Method, but it was also writing "Payment Methods" underneath it, which is also empty.

![Bildschirmfoto 2021-02-02 um 10 41 01](https://user-images.githubusercontent.com/25100967/106581508-3a48dd00-6543-11eb-96df-50e2acec30c7.png)

I do not see any errors in the frontend or backend. When i load the Checkout page i see that it is loading Content from these URLs:
https://checkoutshopper-test.adyen.com/checkoutshopper/assets/js/pub.v2.8015906602993722.[RANDOM-STRING]/fingerprintjs2.js?parentOrigin=https%3A%2F%2Fdomain.tld
https://checkoutshopper-test.adyen.com/checkoutshopper/assets/js/pub.v2.8015906602993722.[RANDOM-STRING]/dfp.1.0.0.js?parentOrigin=https%3A%2F%2Fdomain.tld

With the Version from github the Response is empty, when i readd the store Version i see response with content. All have the status code 200.

**Comment by danielDigitalArt** [comment]
Any Ideas how i can test the new Version?

**Comment by acampos1916** [comment]
Hey @danielDigitalArt, indeed this looks unrelated to the latest version in Github. Could you try re-saving the plugin configuration so that the origin vars are saved in cache and try again?

**Comment by danielDigitalArt** [comment]
@acampos1916 i tried the following:
Update stroe Version to git Version 1.7.0
Click Save in Plugin Settings.
Reload the Frontend, still seeing the double Payment Rows.
Clear Cache, retry the Front Page, still double Payment Rows.
Click on Reinstall Plugin, Save Settings again, still double Payment Rows.
Clear Cache again, still double Payment Rows.

In Both Rows are no Payment Methods available.

**Comment by acampos1916** [comment]
Hey @danielDigitalArt thanks for the info. Just checking, do you think this is related to #133?

**Comment by danielDigitalArt** [comment]
@acampos1916 i can not really reproduce this, the shop only has one shipping method which is the default shipping method. The shipping methods are not needed in the shop system here, which is why it is not visible in the frontend. Yet the Payment Methods still need to be linked to the shipping method which they are, even after the update to the 1.7 version. There are no other restrictions on the shipping method, nor on the payment method via the Risk Manager.

As i have written, the git version is acting wird, as it insert the Payment methods 2 times, one is the nederlands version and one is in english. This only occurs when using the version from git with the tutorial given in the documentation here "Step 2: Install and configure the Shopware plugin". The Settings in the plugin manager for the plugin were not altered or anything, just saved as suggested.

**Comment by acampos1916** [comment]
Gotcha @danielDigitalArt.

Hey @bortefi would you happen to have an idea what could be wrong here?

**Comment by bortefi** [comment]
@acampos1916 
My first thought goes to \AdyenPayment\Subscriber\OrderEmailSubscriber::shouldStopEmailSending
This subscribers listens on Shopware_Modules_Order_SendMail_Send and stops the sending of e-mails to be sent later.

I do believe this is related to #110, as part to resolve that issue, 
the sending of mails is also tackled to be used without the session.

Maybe we can give a higher priority to #110, so it can be picked up in the next sprint?

**Comment by danielDigitalArt** [comment]
The Double Display of the Payment-Title ist still persistent in the released Version 1.7.1, which is why i created a separate Issue for this:
https://github.com/Adyen/adyen-shopware5/issues/135

**Comment by danielDigitalArt** [comment]
I now tested with the new Version in the Shopware Store (1.7.1) generally it is now sending out the Order-Confirmation E-Mail, so im closing this issue now.
